packages/e2e/cypress/integration/e2e/Canvas/canvas.cy.ts

```diff
@@ -0,0 +1,122 @@
+import { BROWSER_PX_DEVIATION, WheelDirection, ZOOM_IN_COUNT, ZOOM_OUT_COUNT, ZOOM_PX_DEVIATION } from "../constants"
+import { coordinatesFromStringPX, zoomFromMatrix } from "../helpers"
+import { canvasModel } from "./Canvas.model"
+import { CONTEXT } from "./context"
+import { checkCanvasPosition } from "./helpers"
+
+const CANVAS_MOVE_POINT_X = 350
+const CANVAS_MOVE_POINT_Y = 350
+
+context(CONTEXT, () => {
+  beforeEach(canvasModel.open)
+
+  const checkCanvasOriginPosition = (x: number, y: number) =>
+    canvasModel
+      .canvasPositionOrigin()
+      .then(coordinatesFromStringPX)
+      .then(([xCoord, yCoord]) => {
+        expect(Number(xCoord)).to.be.closeTo(x, BROWSER_PX_DEVIATION)
+        expect(Number(yCoord)).to.be.closeTo(y, BROWSER_PX_DEVIATION)
+      })
+
+  describe("Initializing", () => {
+    it("Should match default position", () => {
+      checkCanvasPosition(15, -35)
+      checkCanvasOriginPosition(385, 340)
+    })
+  })
+
+  describe("Movements DnD", () => {
+    const matchEndPosition = () => {
+      checkCanvasPosition(115, 65)
+      checkCanvasOriginPosition(285, 240)
+    }
+
+    it("Should move canvas one movement", () => {
+      canvasModel.dnd(CANVAS_MOVE_POINT_X, CANVAS_MOVE_POINT_Y, 450, 450)
+
+      matchEndPosition()
+    })
+
+    it("Should move canvas chain movement", () => {
+      canvasModel.dnd(CANVAS_MOVE_POINT_X, CANVAS_MOVE_POINT_Y, 550, 550)
+      canvasModel.dnd(550, 550, 550, CANVAS_MOVE_POINT_Y)
+      canvasModel.dnd(550, CANVAS_MOVE_POINT_Y, 450, 450)
+
+      matchEndPosition()
+    })
+
+    it("Should move canvas long movement", () => {
+      canvasModel
+        .mouseDown(CANVAS_MOVE_POINT_X, CANVAS_MOVE_POINT_Y)
+        .realMouseMove(550, 550)
```

Comment:
For example, here. It's coordinates and it uses several times, i think it's should be separate